dvc/objects/db/git.py

```diff
@@ -0,0 +1,24 @@
+import logging
+
+from .base import ObjectDB
+
+logger = logging.getLogger(__name__)
+
+
+class GitObjectDB(ObjectDB):
+    """Dummy read-only ODB for uncached objects in external Git repos."""
+
+    def __init__(self, fs, path_info, **config):
+        from dvc.fs.repo import RepoFileSystem
+
+        assert isinstance(fs, RepoFileSystem)
+        super().__init__(fs, path_info)
+
+    def get(self, hash_info):
+        raise NotImplementedError
+
+    def add(self, path_info, fs, hash_info, move=True, **kwargs):
+        raise NotImplementedError
+
+    def gc(self, used, jobs=None):
+        raise NotImplementedError
```

Comment:
So currently for git files we go the `stage() + save()` way for simplicity, but how do you see this odb in the future? Right now we want md5s from git files, but git odb can provide us with sha*s (and even then there are some details to it), so we'll need some translation layer? Or is there a more elegant way?

Reply:
Or maybe we just need to convert them (aka stage()&save() like we do right now) on-the-fly from GitFS to LocalODB, so that we just use the resulting local objects right away? Since that is essentially the translation layer and once these files are converted into our md5-based objs, they are no longer in real git odb (i mean real-real git odb)

Reply:
Yeah I've been thinking about what we can do here, since git files won't be structured like a regular DVC ODB.

In theory though, if we move to using SHA's for git imports (or in general eventually), we could do an complete ODB implementation for translating git blob storage into DVC object storage. And lookups like hashes_exist could be done directly through git instead of using the fs walk/find method we do for other ODBs.